src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,55 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(100, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.removeDeadAnimal(this);
+    }
+  }
+
+  removeDeadAnimal(animal) {
+    Animal.alive = Animal.alive.filter(a => a !== animal);
+  }
```

Comment:
```suggestion
    if (this.health <= 0) {
      Animal.removeDeadAnimal(this);
    }
  }
```
why do you need to create the another method if you have the static one
<img width="623" alt="image" src="https://github.com/mate-academy/js_herbivores_and_carnivores/assets/70220667/9becb1c4-9bc3-44e8-8ba8-1ea48692e18d">